Sources/MetalSprockets/Core/Element.swift

```diff
@@ -1,7 +1,57 @@
 import MetalSprocketsSupport
 
+// MARK: - Element Protocol
+
+/// The fundamental building block of MetalSprockets render graphs.
+///
+/// `Element` is analogous to SwiftUI's `View` protocol. Each element describes
+/// a unit of GPU work — such as a render pass, pipeline configuration, or draw command.
+/// Elements compose together to form a declarative render graph.
+///
+/// ## Creating Custom Elements
+///
+/// Conform to `Element` and implement the `body` property to compose other elements:
```

Comment:
Might be cool to mention stuff like ForEach, Group etc. Link to ElementBuilder too